src/components/UserMenu/UserMenu.tsx

```diff
@@ -21,6 +27,18 @@ type Props = {
 const UserMenu = ({ showPaymentsItem, small = false, onClick }: Props) => {
   const { t } = useTranslation('user');
   const navigate = useNavigate();
+  const { accessModel } = useConfigStore();
+  const { profile: currentProfile } = useProfileStore();
+  const profilesEnabled = useProfilesFeatureEnabled();
+
+  const { data, isFetching } = useListProfiles();
+  const profiles = data?.responseData.collection;
+
+  if (profilesEnabled && !profiles?.length) {
+    unpersistProfile();
+  }
```

Comment:
This logic should be elsewhere (no stores in components). The `unpersistProfile` seems a bit out of place, but that can be just me 😅